app/src/crypto.c

```diff
@@ -522,38 +540,47 @@ __Z_INLINE zxerr_t copyKeys(keys_t *saplingKeys, key_kind_e requestedKeys, uint8
     return zxerr_ok;
 }
 
-zxerr_t crypto_generateSaplingKeys(uint8_t *output, uint16_t outputLen, key_kind_e requestedKey) {
-    if (output == NULL || outputLen < 3 * KEY_LENGTH) {
-        return zxerr_buffer_too_small;
+zxerr_t crypto_computeSaplingSeed(uint8_t *sk) {
```

Comment:
you can change the input param to `uint8_t spendingKey[static KEY_LENGTH]` to check both null and size